lib/compiler.js

```diff
@@ -538,3 +548,33 @@ function resolveRelativePaths(relativePaths, appRootDir) {
     }
   });
 }
+
+function isPreferredExtension (filename, includeExtensions, excludeExtensions) {
```

Comment:
[KISS](http://en.wikipedia.org/wiki/KISS_principle) and remove `includeExtensions` and `excludeExtensions` that are not used AFAICT.